FCBHDBP-436 modify sofria-cli

vichugofsl opened the pull request:
# Description
It has updated the sofria_mediaId.js file to achive to create the .json files from usx directory. It will read each .usx file from a specific directory and it will create a .json file for each chapter.

E.g. I have used the folder Spanish_N2SPNTLA_USX and I have executed the following command:

```shell
node sofria_mediaId.js ~/fcbh-etl-local/Spanish_N2SPNTLA_USX
```
```shell
pattern:
node sofria_mediaId.js [PATH_USX_FOLDER]
```

I have added the parameters: offset and limit to process a limit of files. E.g. about the Spanish_N2SPNTLA_USX folder the two first files are:  001GEN.usx and 002EXO.usx

The output of the above command was:

```shell
## pwd: output/Spanish_N2SPNTLA_USX-json/

001GEN_001.json  001GEN_010.json  001GEN_019.json  001GEN_028.json  001GEN_038.json  001GEN_047.json  002EXO_006.json  002EXO_015.json  002EXO_024.json  002EXO_034.json  002EXO_044.json
001GEN_002.json  001GEN_011.json  001GEN_020.json  001GEN_030.json  001GEN_039.json  001GEN_048.json  002EXO_007.json  002EXO_016.json  002EXO_025.json  002EXO_035.json  002EXO_045.json
001GEN_003.json  001GEN_012.json  001GEN_021.json  001GEN_031.json  001GEN_040.json  001GEN_049.json  002EXO_008.json  002EXO_017.json  002EXO_026.json  002EXO_036.json  002EXO_046.json
001GEN_004.json  001GEN_013.json  001GEN_022.json  001GEN_032.json  001GEN_041.json  001GEN_050.json  002EXO_009.json  002EXO_018.json  002EXO_027.json  002EXO_037.json  002EXO_047.json
001GEN_005.json  001GEN_014.json  001GEN_023.json  001GEN_033.json  001GEN_042.json  002EXO_001.json  002EXO_010.json  002EXO_019.json  002EXO_028.json  002EXO_038.json  002EXO_048.json
001GEN_006.json  001GEN_015.json  001GEN_024.json  001GEN_034.json  001GEN_043.json  002EXO_002.json  002EXO_011.json  002EXO_020.json  002EXO_030.json  002EXO_039.json  002EXO_049.json
001GEN_007.json  001GEN_016.json  001GEN_025.json  001GEN_035.json  001GEN_044.json  002EXO_003.json  002EXO_012.json  002EXO_021.json  002EXO_031.json  002EXO_041.json  002EXO_050.json
001GEN_008.json  001GEN_017.json  001GEN_026.json  001GEN_036.json  001GEN_045.json  002EXO_004.json  002EXO_013.json  002EXO_022.json  002EXO_032.json  002EXO_042.json
001GEN_009.json  001GEN_018.json  001GEN_027.json  001GEN_037.json  001GEN_046.json  002EXO_005.json  002EXO_014.json  002EXO_023.json  002EXO_033.json  002EXO_043.json
```
The above files were created inside folder: `./output/Spanish_N2SPNTLA_USX-json/`

## Issue Link
https://fullstacklabs.atlassian.net/browse/FCBHDBP-436

## How Do I QA This
- Download the local USX files into folder:` ~/fcbh-etl-local/`
- Run: `node sofria_mediaId.js ~/fcbh-etl-local/Spanish_N2SPNTLA_USX 0 2`
- Check that json files are created inside folder: `./output/Spanish_N2SPNTLA_USX-json/`. There must be 66 files

bradflood commented on sofria_mediaId.js:
```diff
@@ -1,44 +1,144 @@
 
-if (process.argv.length !== 5) {
-    throw new Error(`usage: sofria-mediaId <USX FILESET DIRECTORY>`);
+const progressBar = new cliProgress.SingleBar({}, cliProgress.Presets.shades_classic);
```
note that the main use case for sofria-cli will be headless (eg invoked from dbp-etl, and therefore no UI). It's ok to have the progressBar capability as long as it works in a headless environment

vichugofsl replied:
ok I got it. It has been removed.

bradflood commented on sofria_mediaId.js:
```diff
@@ -1,44 +1,144 @@
+    }
+}
+
+async function printChapterContent(usxFileName, chapter) {
```
change function name to generateChapterContent

vichugofsl replied:
It has been updated.

bradflood commented on sofria_mediaId.js:
```diff
@@ -1,44 +1,144 @@
+
+    if (content !== "" && usxFile.suffix !== "") {
+        pk.importDocument({
+            lang: "xxx",
```
change xxx and yyy to empty string

vichugofsl replied:
@bradflood It has been replaced by "eng". E.g.

```javascript
pk.importDocument({
    lang: "eng",
    abbr: ""
}, usxFile.suffix, content);
```
The above because if we set `lang=""`, the Proskomma module will throw the following error:

`Error processing file listValue '' does not match regex '[a-z]{3}'`

bradflood replied:
ok, I didn't realize that.
"eng" will be incorrect almost always. So please change it to xxx

vichugofsl replied:
@bradflood , I have changed to `"xxx"`. See: https://github.com/faithcomesbyhearing/sofria-cli/pull/3/files#diff-26c5163d807f91f16662e922a42e39422ed5b39cb8bb6bb57d7260180a3dc273R75

bradflood commented on sofria_mediaId.js:
```diff
@@ -1,44 +1,144 @@
+        }
+        return listFiles;
+    } catch (err) {
+        console.error('Error to read directory!', path, err);
```
change message to Unable to read directory or filename
and add dirPath to the error message, in case the exception occurs on the readdir statement

vichugofsl replied:
It has been updated.

bradflood commented on sofria_mediaId.js:
```diff
@@ -1,44 +1,144 @@
+    return String(num).padStart(places, '0')
 }
 
+const offset = process.argv.length > 3 ? Number(process.argv[3]) : 0;
```
remove offset and limit. the sofria-cli will always be invoked as one unit. the offset and limit just make it more complex to invoke from within dbp-etl

vichugofsl replied:
It has been removed.

vichugofsl commented:
@bradflood, I have done a refactor to improve the performance. Added the module es6-promise-pool to improve the performance because it was taking more than 30 minutes to perform a folder with 66 USX files. After we have added the es6-promise-pool module, it is taking about 3 minutes to run the same folder with the same 66 USX files.

You can see the implementation in the following two lines:

- https://github.com/faithcomesbyhearing/sofria-cli/pull/3/files#diff-26c5163d807f91f16662e922a42e39422ed5b39cb8bb6bb57d7260180a3dc273R144

- https://github.com/faithcomesbyhearing/sofria-cli/pull/3/files#diff-26c5163d807f91f16662e922a42e39422ed5b39cb8bb6bb57d7260180a3dc273R96

bradflood reviewed (commented):
see comment about lang to xxx
